### ISSUE TITLE
Reified global: false alarm

### ISSUE THREAD

**Issue by erikvader** [erikvader/MiniZinc-linter#33]
Add `redundant_constraint` and the like to `is_conjunctive` in utils.

**Comment by matsc-at-sics-se** [comment]
No, the bug is still there:

```
include "globals.mzn";

set of int: I = {2,3,5};

var 1..3: x;
var 1..3: y;
var 1..3: z;

constraint
  if card(I) > 1 then alldifferent([x,y,z]) else true endif;
```

```
$ lzn foo.mzn 
[...]
/home/matsc/foo.mzn:10.23-14.43: reified global constraint [global-reified(17)]
   |     if card(I) > 1 then alldifferent([x,y,z]) else true endif;
   |                         ^~~~~~~~~~~~~~~~~~~~~
```

**Comment by erikvader** [comment]
so no reification happens if the `if`-part is `par`?

**Comment by Pierre-Flener** [comment]
Indeed not. The flattener even spots a variable-independent `if`-test and replaces the whole expression by its `then` or `else` part, as appropriate.
For the same reason, in #31, it becomes z=y, without needing any reification.

**Comment by matsc-at-sics-se** [comment]
Yet another false alarm, from `minizinc-benchmarks-master/cyclic-rcpsp`:

```
$ lzn -i 1 -i 5 -i 9 -i 18 -i 19 rcmsp.mzn 
/home/matsc/MZN/minizinc-benchmarks-master/cyclic-rcpsp/rcmsp.mzn:123.17: reified global constraint [global-reified(17)]
   |     cumulative(
   |     ^~~~~~~~~~~
```

**Comment by erikvader** [comment]
the `if`-case fixed with 5cfcac4b072c6f0dbc9e78802b370c225fcc5e21
